odoo/models.py

```diff
@@ -774,6 +769,23 @@ def splittor(rs):
                         else:
                             current[i] = False
 
+        # if any xid should be exported, only do so at toplevel
+        if batch_invalidate and any(f[-1] == 'id' for f in fields):
```

Comment:
I don't see the link between the flag `batch_invalidate` and the generation of external ids.
The latter should **always** occur, shouldn't it?

Reply:
Nah, the method is recursive but we want the fetch/gen of external ids (as well as the cache invalidation hence the reuse of that flag) to only happen at the toplevel.

Reply:
So the actual semantics of `batch_invalidate` is `is_toplevel`, right?

Reply:
Yeah.